core/src/main/scala/org/apache/spark/ui/JettyUtils.scala

```diff
@@ -330,7 +330,7 @@ private[spark] object JettyUtils extends Logging {
 
           // redirect the HTTP requests to HTTPS port
           httpConnector.setName(REDIRECT_CONNECTOR_NAME)
-          collection.addHandler(createRedirectHttpsHandler(securePort, scheme))
+          collection.addHandler(createRedirectHttpsHandler(connector, scheme))
```

Comment:
So is there a reason why you didn't make this a single line change to pass the correct port?